### PR TITLE
Close open file handles

### PR DESCRIPTION
**Pull request by dholms**
Couple spots where we weren't closing open file handles. Started popping up in tests

### CODE REVIEW

File: packages/pds/src/image/server.ts [hunk 1]
```diff
@@ -145,6 +146,8 @@ export class BlobDiskCache implements BlobCache {
         throw new BlobNotFoundError()
       }
       throw err
+    } finally {
+      await handle?.close()
```
Comment (devinivy) [inline packages/pds/src/image/server.ts:150]: Naively I would expect this to close the file descriptor in such a way that it prevents consumers from reading the returned stream—do you know why that isn't the case?  Or if that does turn out to be case, maybe this just needs to go in the `catch {}` path?

File: packages/pds/src/storage/disk-blobstore.ts [hunk 2]
```diff
@@ -112,9 +112,13 @@ export class DiskBlobStore implements BlobStore {
...
+    if (!exists) {
+      throw new BlobNotFoundError()
+    }
     try {
-      const handle = await fs.open(this.getStoredPath(cid), 'r')
-      return handle.createReadStream()
+      return fsSync.createReadStream(path)
     } catch (err) {
       if (isErrnoException(err) && err.code === 'ENOENT') {
         throw new BlobNotFoundError()
```
Comment (devinivy) [inline packages/pds/src/storage/disk-blobstore.ts:124]: This error can't really occur here anymore, I don't think—I would expect it to instead be emitted on the returned stream.  That could technically still happen if the file is removed between the `fileExists()` check above and when the file stream is created, though I don't think that is especially likely or a major deal.

File: packages/pds/src/image/server.ts (hunk 1 above)
Comment (dholms) [reply packages/pds/src/image/server.ts]: yeah i think you're right i totally missed that 🙃 

Okay i just used `fsSync` to return the readstream

File: packages/pds/src/storage/disk-blobstore.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/src/storage/disk-blobstore.ts]: yup yup good call 👍 

fixed both of these up & changed my approach just a bit

avoiding open handles in general, just wanna create a read stream & let node handle closing it

**Review by devinivy — APPROVED** [review]
Yup!

File: packages/common/src/fs.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import fs from 'fs/promises'
+import { isErrnoException } from './util'
+
+export const fileExists = async (location: string): Promise<boolean> => {
+  try {
+    await fs.access(location, fs.constants.F_OK)
```
Comment (dholms) [inline packages/common/src/fs.ts:6]: `exists` is deprecated & `existsSync` is well... synchronous. 

`access` is lighterweight than `stat` & gives us what we want

Comment (devinivy) [reply packages/common/src/fs.ts:6]: Ace!

File: packages/pds/src/storage/disk-blobstore.ts (hunk 2 above)
Comment (dholms) [reply packages/pds/src/storage/disk-blobstore.ts]: ohhh got it you're probably right

ah well, this feels a bit more explicit to me so i'll stick with it

Comment (devinivy) [reply packages/pds/src/storage/disk-blobstore.ts]: I think that sounds good 💎

**Pull request merged**